File: frontend/src/app/models/device-data.ts
export interface DeviceData {
  id: string;
  name: string;
  serialNumber: string;
  batteryLevel: number;
  alerts: SingleDeviceRecord[];
  data: SingleDeviceRecord[];
}

export interface SingleDeviceRecord {
  lat: number;
  long: number;
  temp: number | undefined;
  humidity: number | undefined;
  timestamp: string;
}

export interface SingleDeviceAlert {
  code: string;
}

File: backend/src/main/java/com/anyungu/data/mapping/entitiles/DeviceData.java
package com.anyungu.data.mapping.entitiles;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class DeviceData {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, unique = true)
    private Integer id;

    private Float latitude;
    private Float longitude;
    private Float temp;
    private Float humidity;
    private LocalDateTime timestamp;
    private Integer deviceId;
    
}
